z_test_implementation.py

```diff
@@ -0,0 +1,64 @@
+import time
```

Comment:
We should move this file to be an example under examples/ray/...

Reply:
This file is weird for me and I don't know why, but when I put it under examples I get back the complaining from ray that it tries to add 1s to a ray reference, i.e. as if it were nested again. If I run it where it is, everything is fine and works as intended.